src/Services/RetrieveExternalDataService/RetrieveExternalDataService.cs

```diff
@@ -40,10 +44,16 @@ public async Task Process(List<IAction> actions, FormSchema formSchema, string f
             foreach (var action in actions)
             {
                 var response = new HttpResponseMessage();
-                var entity = _actionHelper.GenerateUrl(action.Properties.URL, mappingData.FormAnswers);
+                var submitSlug = action.Properties.PageActionSlugs.FirstOrDefault(_ =>
+                    _.Environment.Equals(_environment.EnvironmentName.ToS3EnvPrefix()));
```

Comment:
could we add a toLower to this check on both, to remove any possibility of case issues.